src/twisted/python/components.py

```diff
@@ -337,7 +337,7 @@ def __init__(self, original):
     for name in iface:
         contents[name] = _ProxyDescriptor(name, originalAttribute)
     proxy = type("(Proxy for {})".format(reflect.qual(iface)), (object,), contents)
-    declarations.classImplements(proxy, iface)
+    declarations.classImplements(proxy, iface)  # type: ignore[misc]
```

Comment:
```suggestion
    # mypy-zope declarations.classImplements only works when passing
    # a concrete class type
    declarations.classImplements(proxy, iface)  # type: ignore[misc]
```

Reply:
I ran into this recently and filed [this PR against mypy-zope](https://github.com/Shoobx/mypy-zope/issues/50); not enturely sure if it's a mypy problem or a mypy-zope problem.